src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/index_add_scatter_elements_update.cpp

```diff
@@ -0,0 +1,293 @@
+        Output<Node> rank = std::make_shared<opset10::ShapeOf>(shape, output_type);
+        if (as_scalar) {
+            auto axis_0 = opset10::Constant::create(output_type, Shape{}, {0});
+            rank = std::make_shared<opset10::Squeeze>(rank, axis_0);
+        }
+        return std::make_tuple(shape, rank);
+    }
+
+    std::shared_ptr<Node> get_node_axes_range(const Output<Node>& x) {
+        auto start = std::make_shared<opset10::Constant>(element::i32, Shape{}, 0);
+        auto step = std::make_shared<opset10::Constant>(element::i32, Shape{}, 1);
+        Output<Node> reduced_rank;
+        std::tie(std::ignore, reduced_rank) = get_shape_rank(x, true);
+        return std::make_shared<opset10::Range>(start, reduced_rank, step, element::i32);
+    }
+
+    size_t normalized_axis;  // normalized_axis
+};
+
+TEST_P(IndexAddTest, CompareWithRefs) {
+    run();
+}
+
+namespace {
+// map<inputShape, map<updatesShape, axis>>
+std::map<std::vector<size_t>, std::map<std::vector<size_t>, std::vector<int>>> axesShapeInShape {
+    {{3}, {{{2}, {0, -1}}, {{3}, {0, -1}}/*, {{0}, {0, -1}}*/}}, // TODO: empty tensor failing in template plugin
+    {{4, 6}, {{{3, 6}, {0, -2}}, {{4, 6}, {0, 1, -1}}/*, {{0, 2}, {0, -2}}*/}},  // axis 0
+    {{2, 4}, {{{2, 3}, {1, -1}}, {{2, 4}, {0, 1, -1}}/*, {{4, 0}, {1, -1}}*/}},  // axis 1
+    {{1, 4096}, {{{1, 4096}, {0}}}},
+    {{32, 4096}, {{{16, 4096}, {0}}, {{32, 4096}, {0}}}},
+    {{1024, 4096}, {{{512, 4096}, {0}}, {{1024, 4096}, {0}}}},
+    {{8192, 4096}, {{{4096, 4096}, {0}}}},
+};
+
+inline std::vector<InputShape> partial_shapes_to_test_representation(
+    const std::vector<ov::PartialShape>& shapes) {
+    std::vector<InputShape> result;
+    for (const auto& staticShape : shapes) {
+        result.push_back({{staticShape}, {staticShape.get_shape()}});
+    }
+    return result;
+}
+
+std::vector<ov::test::InputsAndAxis> combine_shapes(
+    const std::map<std::vector<size_t>, std::map<std::vector<size_t>, std::vector<int>>>& input_shapes) {
+    std::vector<ov::test::InputsAndAxis> res_vec;
+    for (auto& input_shape : input_shapes) {
+        for (auto& item : input_shape.second) {
+            for (auto& elt : item.second) {
+                res_vec.push_back(ov::test::InputsAndAxis{
+                    partial_shapes_to_test_representation({ov::PartialShape(input_shape.first), ov::PartialShape(item.first)}),
+                    elt});
+            }
+        }
+    }
+    return res_vec;
+}
+
+INSTANTIATE_TEST_SUITE_P(smoke_IndexAddTest,
+                         IndexAddTest,
+                         ::testing::Combine(::testing::ValuesIn(combine_shapes(axesShapeInShape)),
+                                            ::testing::Values(v12::ScatterElementsUpdate::Reduction::SUM, v12::ScatterElementsUpdate::Reduction::NONE),
+                                            ::testing::Values(ElementType::f32, ElementType::bf16), // data precision
```

Comment:
Did you check that bf16 type ends up in the plugin. If I remember that correctly, the convert precision transformation will insert bf16->f32 convert layers around the ScatterUpdate layer. It's better to use config to enforce bf16 case.

Reply:
yes, I confirmed that it is bf16 data type for SEU nodes. There is no convert node before it, as it is bf16 beginning from corresponding model inputs.

Reply:
Actually the reference impl of SEU-12 does not support bf16. Template plugin will convert to float inside.
From a subgraph perspective (torch.index_add), if cpu plugin yields the same outputs as reference for given inputs, that's okay.
So we can depend on the default config of precision enforce, and I update here to test with model level precisions.